paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -323,15 +386,72 @@ def instances_deployed(cluster, service, instances, git_sha):
                              .format(service, instance, cluster,
                                      status.marathon.running_instance_count,
                                      status.marathon.expected_instance_count))
+                instances_in.task_done()
+                instances_out.put(instance)
                 continue
             paasta_print("Complete: {}.{} on {} looks 100% deployed at {} "
                          "instances on {}"
                          .format(service, instance, cluster,
                                  status.marathon.running_instance_count,
                                  status.git_sha))
-            result.append(instance)
-
-    return result
+            instances_in.task_done()
+
+
+def _query_clusters(cluster_map, service, git_sha, green_light):
+    """Run _run_cluster_worker() in a separate thread for each paasta cluster
+
+    :param cluster_map: see paasta_tools.generate_deployments_for_service.get_cluster_instance_map_for_service
+    :param cluster: the name of the cluster.
+    :param service: the name of the service.
+    :param git_sha: git sha marked for deployment.
+    :param green_light: an instance of threading.Event().
+                        It is supposed to be cleared when KeyboardInterrupt is
+                        received. All running threads should check it
+                        periodically and exit when it is cleared.
+    """
+    for cluster, cluster_data in cluster_map.items():
+        if cluster_data['deployed'] != len(cluster_data['instances']):
+            worker = Thread(target=_run_cluster_worker,
+                            args=(cluster, cluster_data, service, git_sha,
+                                  green_light))
+            worker.start()
+            cluster_data['worker'] = worker
+
+    for cluster, cluster_data in cluster_map.items():
+        try:
+            while green_light.is_set() and cluster_data['worker'].isAlive():
+                time.sleep(.2)
+        except (KeyboardInterrupt, SystemExit):
+            green_light.clear()
+            paasta_print('KeyboardInterrupt received. Terminating..')
+        cluster_data['worker'].join()
```

Comment:
do we need timeouts for this and the other ``join``, given they'll block indefinitely?